connexion/decorators/uri_parsing.py

```diff
@@ -188,7 +188,7 @@ def _make_deep_object(k, v):
         root_key = k.split("[", 1)[0]
         if k == root_key:
             return (k, v, False)
-        key_path = re.findall(r'\[(\w+)\]', k)
+        key_path = re.findall(r'\[(\S[^\[\]]*)\]', k)
```

Comment:
Also, this changes the logic in some cases. For example:

```
In [14]: re.match(r'\[(\w+)\]', '[[x]]')                                                                              

In [15]: re.match(r'\[(\S[^\[\]]*)\]', '[[x]]')                                                                       
Out[15]: <re.Match object; span=(0, 4), match='[[x]'>

In [16]: re.match(r'\[(\S[^\[\]]*)\]', '[[x]]').group(1)                                                              
Out[16]: '[x'
```

I assume that either `"x"` or `"[x]"` is intended to be a key in this case, not `"[x"`.

Reply:
Similar with `[     ]` — a space, many spaces, or any space-starting string cannot be a key — is it intended?

Reply:
spaces are fixed. I agree that the other case is not intended, but i'm lacking time/interest to invest the time to make the regex bullet proof for that now.